### PR TITLE
Java: Add Android Gradle Plugin 8 and Spring Boot 3 tests

### PR DESCRIPTION
**Pull request by smowton**

### CODE REVIEW

File: java/integration-tests-lib/toolchains_test_utils.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import os.path
+import sys
+import tempfile
+
+def actions_expose_all_toolchains():
```
Comment (mbg) [inline java/integration-tests-lib/toolchains_test_utils.py:5]: This is a really nice addition for testing!

**Review by mbg — COMMENTED** [review]
Reviewed this as part of the associated, internal PR. Nice to have some integration tests for this! Couple of questions:

- The two new integration tests include the `.gradle` folder. Is that intentional? If so why? Other integration tests that use Gradle don't seem to do this.
- It may be useful to add a comment before the use of `try_use_java11()` to note why this is necessary. Particularly, it's not immediately obvious what impact this has on the version selection logic and how it affects the test outcome when compared to what may be happening in users' workflows.

File: java/ql/integration-tests/all-platforms/java/android-8-sample/.gitignore
Comment (mbg) [inline java/ql/integration-tests/all-platforms/java/android-8-sample/.gitignore:1]: Minor: bit odd to have such a comprehensive `.gitignore` specifically for this test. Consider moving it up to the most distant ancestor of this directory where it makes sense.

**Review by mbg — APPROVED** [review]
Thanks for addressing my comments. This seems fine as is now, with one minor suggestion for improvement, but up to you if you want to act on it here / in a different PR / at all.

File: java/ql/integration-tests/all-platforms/java/android-8-sample/.gitignore
Comment (smowton) [reply java/ql/integration-tests/all-platforms/java/android-8-sample/.gitignore:1]: These things are generated by Gradle wrapper installation-- you can see quite a few in the test directories for this reason

**Pull request merged**